subsys/usb/device/usb_descriptor.c

```diff
@@ -45,6 +45,7 @@ struct common_descriptor {
 #define USB_DESC_MANUFACTURER_IDX			1
 #define USB_DESC_PRODUCT_IDX				2
 #define USB_DESC_SERIAL_NUMBER_IDX			3
+#define USB_DESC_CONFIGURATION_IDX			4
```

Comment:
I don't like this pattern at all. The string descriptors are optional in most cases, and by doing this this way the default sample applications are going to be bloated with some useless information IMHO.

Reply:
What are you suggesting to move forward? A Kconfig option to remove the code entirely if not needed?

Reply:
I have no idea why would anyone want to have such meaningless USB configuration string in the first place.

I agree that there should be some mechanism for application to be able to provide optional string descriptors. But if someone wants to implement optional string descriptors (there are a bunch of these, virtually every descriptor field that starts with `i` is optional string descriptor) then the string descriptor should not be available by default (i.e. to have its value set to 0).

Reply:
The chromebook "hammer" detects the USB configuration (which includes the firmware version string)[link](https://source.chromium.org/chromiumos/chromiumos/codesearch/+/main:src/platform2/hammerd/update_fw.cc;drc=a9f2a2a173b331ad350c6cd0a764d46810d5ad51;l=183) [link](https://source.chromium.org/chromiumos/chromiumos/codesearch/+/main:src/platform2/hammerd/usb_utils.cc;drc=17482269cd786dbb14961c83c85a183f28061ac2;l=112) to determine whether the keyboard dock firmware (running on ZephyrOS) needs updating. This is why we created this PR to add support for the USB configuration descriptor. Following Fabio's suggestion, I've added a kernel option to enable USB configuration descriptor availability.

Reply:
Please remove "Tested with: Google spikyrock project" from the commit message.